From 4db11448e83f96524d757131bd465bdf1824de4d Mon Sep 17 00:00:00 2001
From: Pawel Zalewski <pzalewski@thegoodpenguin.co.uk>
Date: Tue, 28 Nov 2023 08:18:20 +0000
Subject: [PATCH 08/10] duk_config: use debug and self test

Signed-off-by: Pawel Zalewski <pzalewski@thegoodpenguin.co.uk>
---
 src/libs/zbxembed/duk_config.h | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/src/libs/zbxembed/duk_config.h b/src/libs/zbxembed/duk_config.h
index d4c036eb858..da0ae790372 100644
--- a/src/libs/zbxembed/duk_config.h
+++ b/src/libs/zbxembed/duk_config.h
@@ -2936,7 +2936,7 @@ typedef struct duk_hthread duk_context;
 #undef DUK_USE_DATE_GET_NOW
 #undef DUK_USE_DATE_PARSE_STRING
 #undef DUK_USE_DATE_PRS_GETDATE
-#undef DUK_USE_DEBUG
+#define DUK_USE_DEBUG
 #undef DUK_USE_DEBUGGER_DUMPHEAP
 #undef DUK_USE_DEBUGGER_INSPECT
 #undef DUK_USE_DEBUGGER_PAUSE_UNCAUGHT
@@ -2944,8 +2944,8 @@ typedef struct duk_hthread duk_context;
 #define DUK_USE_DEBUGGER_THROW_NOTIFY
 #undef DUK_USE_DEBUGGER_TRANSPORT_TORTURE
 #define DUK_USE_DEBUG_BUFSIZE 65536L
-#define DUK_USE_DEBUG_LEVEL 0
-#undef DUK_USE_DEBUG_WRITE
+#define DUK_USE_DEBUG_LEVEL 1
+#define DUK_USE_DEBUG_WRITE
 #define DUK_USE_DOUBLE_LINKED_HEAP
 #define DUK_USE_DUKTAPE_BUILTIN
 #define DUK_USE_ENCODING_BUILTINS
-- 
2.47.1

